src/parser_factory.cpp

```diff
@@ -120,7 +120,8 @@ std::shared_ptr<ParserInterface>
         {
             logging::logMessage("KWD vpd parser selected for VPD file path " +
                                 i_vpdFilePath);
-            return std::make_shared<KeywordVpdParser>(i_vpdVector);
+            return std::make_shared<KeywordVpdParser>(i_vpdVector,
```

Comment:
validated at the caller's place here https://github.com/ibm-openbmc/openpower-vpd-parser/blob/P11_Dev/src/manager.cpp#L218 . But yeah we have to validate in this API as well so that we are safe even if anyother caller failed to validate.

Reply:
we will create a new issue to address such validations across the repo.